frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Outlet } from 'react-router-dom';
+
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { useEffect, useNavigate } from '~/libs/hooks/hooks.js';
+import { storage, StorageKey } from '~/libs/packages/storage/storage.js';
+import { type UserGetAllItemResponseDto } from '~/packages/users/users.js';
+
+const ProtectedRoute = (): React.ReactElement => {
+  const navigate = useNavigate();
+  const { users } = useSelector(
+    (user: { users: UserGetAllItemResponseDto[]; dataStatus: string }) => user,
+  );
```

Comment:
1. You should use `useAppSelector` from common hooks
2. `dataStatus` has it's own typing (not just a `string`)